app/controllers/reservations_controller.rb

```diff
@@ -1,2 +1,63 @@
 class ReservationsController < ApplicationController
+  # Get All Reservations With Status Pending
+  def pending_reservation_admin
+    if admin?
+      pending_reservations = Reservation.all.where(status: 'Pending')
+      render json: pending_reservations, each_serializer: ReservationSerializer
+    else
+      render json: { error: 'Not Allowed' }, status: :unauthorized
+    end
+  end
+
+  # Update a Reservation Based On Admin Decision
+  def handle_pending_reservation_admin
+    if admin?
+      reservation = Reservation.find_by(id: action_params[:reservation_id].to_i)
+      raise ActiveRecord::RecordNotFound unless reservation
+
+      case action_params[:decision]
+      when 'Confirm'
+        accept_reservation(reservation)
+      when 'Refuse'
+        refuse_reservation(reservation)
+      end
+    else
+      render json: { error: 'Not Allowed' }, status: :unauthorized
+    end
+  end
+
+  private
+
+  # If Admin Approve Update reservation status
+  def accept_reservation(reservation)
+    reservation.status = 'Confirmed'
+    if reservation.save
+      create_notification(reservation)
+      render json: reservation, each_serializer: ReservationSerializer
+    else
+      render json: { error: 'Bad Request' }, status: :not_acceptable
+    end
+  end
+
+  # If Admin Refuse Update reservation status
+  def refuse_reservation(reservation)
+    reservation.status = 'Refused'
+    if reservation.save
+      create_notification(reservation)
+      render json: reservation, each_serializer: ReservationSerializer
+    else
+      render json: { error: 'Bad Request' }, status: :not_acceptable
+    end
+  end
```

Comment:
These two methods are almost the same except the `status`. What do you think about something like this?
```ruby
def update_reservation(reservation, status)
    reservation.status = status
    if reservation.save
      create_notification(reservation)
      render json: reservation, each_serializer: ReservationSerializer
    else
      render json: { error: 'Bad Request' }, status: :not_acceptable
    end
end
```

And then we can call it like this

```ruby
update_reservation(reservation, 'Refused')
```
and
```ruby
update_reservation(reservation, 'Confirmed')
```